.github/workflows/reuse.yaml

```
@@ -0,0 +1,134 @@
+        if: ${{ inputs.tag == 'image'}}
+        run: git fetch --prune --unshallow --tags
+
+      - name: Set up Go
+        uses: actions/setup-go@424fc82d43fa5a37540bae62709ddcc23d9520d4 # v2.1.5
+        with:
+          go-version: 1.16
+
+      - name: Install Cosign
+        uses: sigstore/cosign-installer@116dc6872c0a067bcb78758f18955414cdbf918f # v1.4.1
+        with:
+          cosign-release: 'v1.4.1'
+
+      - name: Cache Go modules
+        uses: actions/cache@d9747005de0f7240e5d35a68dca96b3f41b8b340 # v1.2.0
+        with:
+          path: ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+
+      - name: Refresh Go report card
+        if: ${{ inputs.tag == 'release'}}
+        uses: creekorful/goreportcard-action@1f35ced8cdac2cba28c9a2f2288a16aacfd507f9 # v1.0
+
+      - name: Log into ghcr.io
+        uses: docker/login-action@master
+        with:
+          registry: ghcr.io
+          username: ${{secrets.registry_username}}
+          password: ${{secrets.registry_password}}
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@4ab11c41e45d028884a99163086648e898eed25 # v1.6.0
+        id: buildx
+        with:
+          install: true
+
+      - name: Set Version
+        if: ${{ inputs.tag == 'release'}}
+        run: |
+          echo "KYVERNO_VERSION=$(git tag --list --sort=-version:refname "v[0-9]*" | head -n 1)" >> $GITHUB_ENV
+
+      - name: Generate SBOM JSON
+        if: ${{ inputs.image_name == 'kyverno' && inputs.tag == 'release'}}
+        uses: CycloneDX/gh-gomod-generate-sbom@v1
+        with:
+          version: v1
+          args: mod -licenses -json -output kyverno-${{ env.KYVERNO_VERSION }}-bom.cdx.json
+          
+      - name: Upload SBOM JSON
+        if: ${{ inputs.image_name == 'kyverno' && inputs.tag == 'release'}}
+        uses: actions/upload-artifact@v2
+        with:
+          name: kyverno-bom-cdx
+          path: kyverno-v*-bom.cdx.json
+
+      - name : Docker images publish
+        run: make ${{inputs.publish_command}}
+
+      - name: get digest
+        id: get-step
+        run: |
+          echo "::set-output name=digest::$(make ${{inputs.digest_command}})"
+
+      - name: Sign image
+        env:
+          COSIGN_EXPERIMENTAL: "true"
+          COSIGN_REPOSITORY: "ghcr.io/kyverno/signatures"
+        run: |          
+          cosign sign \
+          -a "repo=${{ github.repository }}" \
+          -a "workflow=${{ github.workflow }}" \
+          -a "ref=${{ github.sha }}" \
+          ghcr.io/kyverno/${{inputs.image_name}}@sha256:${{ steps.get-step.outputs.digest }}
+
+      - name : Sign SBOM
+        if: ${{ inputs.image_name == 'kyverno' && inputs.tag == 'release'}}
+        env:
+          COSIGN_REPOSITORY: "ghcr.io/kyverno/sbom"
+        run: cosign attach sbom --sbom ./*-bom.cdx.json --type cyclonedx  ghcr.io/kyverno/kyverno:${{env.KYVERNO_VERSION}}
```

```suggestion
        run: cosign attach sbom --sbom ./*-bom.cdx.json --type cyclonedx  ghcr.io/kyverno/${{inputs.image_name}}@sha256:${{ steps.get-step.outputs.digest }}
```
Same as above. Also pinning to the digest instead of the tag is probably for the better.
Ultimately the sbom is attached via the digest ref and if cosign is given a tag it resolves it to a digest. By providing a digest we can be sure that nothing is tampered in the middle if there are multiple pushes to the same version happening for some reason.